proxy/src/outbound.rs

```diff
@@ -54,6 +54,60 @@ impl<B> Outbound<B> {
             bind_timeout,
         }
     }
+
+    const DEFAULT_PORT: Option<u16> = Some(80);
+
+    /// TODO: Return error when `HostAndPort::normalize()` fails.
+    /// TODO: Use scheme-appropriate-default
+    fn normalize(authority: &http::uri::Authority) -> Option<HostAndPort> {
+        HostAndPort::normalize(authority, Self::DEFAULT_PORT).ok()
+    }
+
+    /// Determines the logical host:port of the request.
+    ///
+    /// 1. If the parsed URI includes an authority, use that.
+    /// 2. Otherwise, try to load the authority from the `Host` header.
+    ///
+    /// The port is either parsed from the authority or a default of 80 is used.
+    fn host_port(req: &http::Request<B>) -> Option<HostAndPort> {
+        req.uri()
+            .authority_part()
+            .and_then(Self::normalize)
+            .or_else(|| {
+                h1::authority_from_host(req)
+                    .and_then(|h| Self::normalize(&h))
+            })
+    }
+
+    /// Determines the destination to use in the router.
+    ///
+    /// 1. If the request was destined to a logic (named) host, route it by the logical
```

Comment:
s/logic/logical/.

I don't think "named" is the best way to describe this because it might not be named; i.e. it might be an IP address.

I would say something like "If the request explicitly indicates the authority then that authority is used for routing."

NIT: numbering these as "1." and "2." isn't useful.